cmake/Modules/MacroQbtCommonConfig.cmake

```diff
@@ -0,0 +1,106 @@
+# Set common variables and create some interface-only library targets
+# that some or all other targets will link to, either directly or transitively,
+# to consume common compile options/definitions
+
+macro(qbt_common_config)
+
+    # treat value specified by the CXX_STANDARD target property as a requirement by default
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+    # these definitions are only needed for calls to
+    # lt::generate_fingerprint and for the qbittorrent.rc file on Windows
+    add_library(qbt_version_definitions INTERFACE)
+
+    target_compile_definitions(qbt_version_definitions INTERFACE
+        QBT_VERSION_MAJOR=${qBittorrent_VERSION_MAJOR}
+        QBT_VERSION_MINOR=${qBittorrent_VERSION_MINOR}
+        QBT_VERSION_BUGFIX=${qBittorrent_VERSION_PATCH}
+        QBT_VERSION_BUILD=${qBittorrent_VERSION_TWEAK}
+    )
+
+    add_library(qbt_common_cfg INTERFACE)
+
+    # Full C++ 14 support is required
```

Comment:
How do you decide to put what features in the list? I mean is there a compiled list or something like that?

Reply:
@Chocobo1 

> How do you decide what to put in the list?

I went through https://cmake.org/cmake/help/v3.16/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#individual-features-from-c-14 and put in all those that had to do with C++14.

> I mean is there a compiled list or something like that?

Recently I submitted a PR to improve CMake's documentation in this regard: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5151/diffs.
The new result can be seen here, with features properly sorted by C++ standard:

https://cmake.org/cmake/help/git-master/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#high-level-meta-features-indicating-c-standard-support
https://cmake.org/cmake/help/git-master/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#low-level-individual-compile-features
https://cmake.org/cmake/help/git-master/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#individual-features-from-c-14

Reply:
@Chocobo1 Just after typing this, I reviewed the sorted list in https://cmake.org/cmake/help/git-master/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#individual-features-from-c-14 and realized I actually missed a couple. Also I took the opportunity to sort them all alphabetically after `cxx_std_14`. Fixed.

Reply:
>Just after typing this, I reviewed the sorted list in https://cmake.org/cmake/help/git-master/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#individual-features-from-c-14 and realized I actually missed a couple. Also I took the opportunity to sort them all alphabetically after cxx_std_14. Fixed.

You should add an URL link to the cmake document.

Reply:
fixed.